From 526ac839a2c241dee6ff0d5a3375be557bd21c17 Mon Sep 17 00:00:00 2001
From: Johannes Thoma <johannes@johannesthoma.com>
Date: Wed, 9 Jan 2019 19:18:15 +0100
Subject: [PATCH 05/13] Debugging int13 san block device logic.

At that point in time capacity of the san device is 0.
---
 src/arch/x86/interface/pcbios/int13.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/src/arch/x86/interface/pcbios/int13.c b/src/arch/x86/interface/pcbios/int13.c
index c1042b3f..13bbfd8b 100644
--- a/src/arch/x86/interface/pcbios/int13.c
+++ b/src/arch/x86/interface/pcbios/int13.c
@@ -395,10 +395,17 @@ static int int13_guess_geometry ( struct san_device *sandev, void *scratch ) {
 	if ( ! int13->cylinders ) {
 		/* Avoid attempting a 64-bit divide on a 32-bit system */
 		blocks = int13_capacity32 ( sandev );
+DBGC ( sandev , "blocks is %d\n", blocks);
+DBGC ( sandev , "sandev_capacity() is %lld\n", sandev_capacity(sandev) );
+DBGC ( sandev , "sandev->capacity.blocks is %lld\n", sandev->capacity.blocks);
+
 		blocks_per_cyl = ( int13->heads * int13->sectors_per_track );
+DBGC ( sandev , "blocks_per_cyl is %d\n", blocks_per_cyl);
 		assert ( blocks_per_cyl != 0 );
 		int13->cylinders = ( blocks / blocks_per_cyl );
-		if ( int13->cylinders > 1024 || int13->cylinders == 0)
+DBGC ( sandev , "int13->cylinders is %d\n", int13->cylinders);
+		/* if ( int13->cylinders > 1024 || int13->cylinders == 0) */
+		if ( int13->cylinders > 1024 )
 			int13->cylinders = 1024;
 	}
 
-- 
2.17.0

